tsl/src/chunk_api.c

```diff
@@ -1627,3 +1629,81 @@ chunk_api_get_chunk_relstats(PG_FUNCTION_ARGS)
 {
 	return chunk_api_get_chunk_stats(fcinfo, false);
 }
+
+Datum
+chunk_api_copy_data(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	const char *src_node_name = PG_ARGISNULL(1) ? NULL : PG_GETARG_CSTRING(1);
+	const char *dst_node_name = PG_ARGISNULL(2) ? NULL : PG_GETARG_CSTRING(2);
+	Chunk *chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	Cache *hcache = ts_hypertable_cache_pin();
+	Hypertable *ht =
+		ts_hypertable_cache_get_entry(hcache, chunk->hypertable_relid, CACHE_FLAG_NONE);
+	List *src_node;
+	List *dst_node;
+	char *src_conn_string;
+	char *operation_id;
+
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	PreventInTransactionBlock(true, get_func_name(FC_FN_OID(fcinfo)));
+
+	if (src_node_name == NULL || dst_node_name == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("data node name cannot be NULL")));
+
+	if (dist_util_membership() != DIST_MEMBER_ACCESS_NODE)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("function must be run on the access node only")));
+
+	if (!hypertable_is_distributed(ht))
+		ereport(ERROR,
+				(errcode(ERRCODE_TS_HYPERTABLE_NOT_DISTRIBUTED),
+				 errmsg("hypertable \"%s\" is not distributed",
+						get_rel_name(chunk->hypertable_relid))));
+
+	ts_cache_release(hcache);
+
+	/* todo: must be a superuser to create publication? */
+
+	src_node = lappend(NIL, pstrdup(src_node_name));
+	dst_node = lappend(NIL, pstrdup(dst_node_name));
+
+	/* prepare shared operation id name for publication and subscription */
+	operation_id = psprintf("ts_copy_chunk_%s",
+							quote_qualified_identifier(NameStr(chunk->fd.schema_name),
+													   NameStr(chunk->fd.table_name))),
+
+	/* create publication on the source data node */
+		ts_dist_cmd_run_on_data_nodes(psprintf("CREATE PUBLICATION %s FOR TABLE %s",
```

Comment:
I am wondering if we should wrap these remote SQL commands in four API functions to be executed on the DNs:

```
chunk_create_subscription()
chunk_drop_subscription()
chunk_create_publication()
chunk_drop_publication()
```

Then a this  `copy_chunk` function is executed on the AN  and will call the above functions.

Why wrap in functions? I imagine that it makes testing easier, we can to parameter validation, and raise more appropriate errors for our use case.

Or is this over-doing it?